### ISSUE TITLE
LSTM validation loss plateau

### ISSUE THREAD

**Issue by YL92** [keras-team/keras#10657]
Hi all,

I am running a lstm with the Tensorflow backend, trying to predict one-day ahead quantiles of stock returns. Unfortunately, while my validation error is low, it does not improve at all over the epochs.

Even when trying to make the model overfit, e.g. with no dropout and stacking of many layers, the picture is more or less the same. I'm wondering if maybe my model is misspecified in some way? The full code can be found below (written in R).

![rplot001](https://user-images.githubusercontent.com/41145967/42622898-c25b1f6e-85c1-11e8-8b0f-ade2ec2a8a61.png)

```

tilted <- function( y_true, y_pred) {
  k_mean(k_maximum(q*(y_true-y_pred), (q-1)*(y_true-y_pred)))
}
generator <- function(data, lookback = FLAGS$lookback, delay = FLAGS$delay, min_index, max_index,
                      shuffle = FALSE, batch_size = FLAGS$batch_size, step = FLAGS$step, test_gen = FALSE) {
  if (is.null(max_index))
    max_index <- nrow(data) - delay - 1
  i <- min_index + lookback
  function() {
    if (shuffle) {
      rows <- sample(c((min_index+lookback):max_index), size = batch_size)
    } else {
      if (i + batch_size >= max_index)
        i <<- min_index + lookback
      rows <- c(i:min(i+batch_size-1, max_index))
      i <<- i + length(rows)
    }
    
    samples <- array(0, dim = c(length(rows), 
                                lookback / step,
                                dim(data)[[-1]]))
    targets <- array(0, dim = c(length(rows)))
    
    for (j in 1:length(rows)) {
      indices <- seq(rows[[j]] - lookback, rows[[j]]-1, 
                     length.out = dim(samples)[[2]])
      samples[j,,] <- data[indices,]
      targets[[j]] <- data[rows[[j]] + delay,1 ]
    }            
    if (test_gen == FALSE) {
    list(samples, targets)
      } else {
        list(samples)
      }
  }
}
train_gen <- generator(
  data,
  lookback = FLAGS$lookback,
  delay = FLAGS$delay,
  min_index = 1,
  max_index = as.integer(nrow(data)*FLAGS$training_data),
  shuffle = FALSE,
  step = FLAGS$step, 
  batch_size = FLAGS$batch_size,
  test_gen = FALSE
)

val_gen = generator(
  data,
  lookback = FLAGS$lookback,
  delay = FLAGS$delay,
  min_index = as.integer(nrow(data)*FLAGS$training_data)+1,
  max_index = as.integer(nrow(data)*(FLAGS$training_data+FLAGS$validation_data)),
  step = FLAGS$step,
  batch_size = FLAGS$batch_size,
  test_gen = FALSE
)

test_gen <- generator(
  data,
  lookback = FLAGS$lookback,
  delay = FLAGS$delay,
  min_index = as.integer(nrow(data)*(FLAGS$training_data+FLAGS$validation_data))+1,
  max_index = NULL,
  step = FLAGS$step,
  batch_size = FLAGS$batch_size,
  test_gen = TRUE
)

train_steps <- as.integer(nrow(data)*FLAGS$training_data)/FLAGS$batch_size

val_steps <- ((as.integer(nrow(data)*(FLAGS$training_data+FLAGS$validation_data)))-
(as.integer(nrow(data)*FLAGS$training_data)+1))/FLAGS$batch_size

test_steps <- ((nrow(data)-delay-1)-(as.integer(nrow(data)*(FLAGS$training_data+FLAGS$validation_data))+1))/FLAGS$batch_size

q = FLAGS$quantile

model_lstm <- keras_model_sequential() 

model_lstm %>% 
  layer_lstm(
    units = FLAGS$neurons1,
    recurrent_dropout = FLAGS$recurrent_dropout,
    return_sequences = if_else(FLAGS$stacking1==TRUE, TRUE, FALSE),
    input_shape = list(NULL, dim(data)[[-1]])
  )

if(FLAGS$stacking1) {
  model_lstm %>%
    layer_lstm(
      units = FLAGS$neurons2,
      recurrent_dropout = FLAGS$recurrent_dropout,
      return_sequences = if_else(FLAGS$stacking2==TRUE, TRUE, FALSE)
    )
}

if(FLAGS$stacking2) {
  model_lstm %>%
    layer_lstm(
      units = FLAGS$neurons3,
      recurrent_dropout = FLAGS$recurrent_dropout,
      return_sequences = FALSE
    )
}

model_lstm %>% layer_dense(units = 1)

model_lstm %>% compile(
  optimizer = optimizer_adam(),
  loss = tilted
)

history_lstm <- model_lstm %>% fit_generator(
  train_gen,
  steps_per_epoch = train_steps,
  epochs = FLAGS$epochs,
  validation_data = val_gen,
  validation_steps = val_steps
)

```
Thanks!
